### ISSUE TITLE
Add function to extract alternate allele dosage estimates from a GDS file

### ISSUE THREAD

**Issue by broomej** [Blue-Lab/blPipeline#5]
Creating a parallel issue to Blue-Lab/assoc_pipeline#59. After discussion, we'll want to add the R function which will give us some time to think about how or if we want to implement a wrapper for this. Should take GDS, chromosome number, start BP and end BP as inputs. I think the output we want is a dataframe with sample.id as the identifier and a variable for each RSID of interest

**Comment by broomej** [comment]
@ElizabethBlue I'm looking at the example code you made for this and you manually set the colnames to be the rsIDs of interest. It doesn't look like the rsIDs are in the annotation field of our GDSs. Instead, they're formatted like this: chr14:73192712:G:C.

In my first pass adding this function, I just use these IDs and format them with `make.names()`, and use that as the variable names. Does that sound like an OK solution? Or is there an easy way I could set the variable names to the rsIDs that I'm not thinking of?

**Comment by ElizabethBlue** [comment]
The rsID info came from me, I think the chr:pos:ref:alt format works for a variant-id that carries some info about the variant instead of just an integer. 
I think we could get a lookup for rsIDs from dbSNP, but it isn't necessary

**Comment by ElizabethBlue** [comment]
https://ftp.ncbi.nih.gov/snp/

**Comment by broomej** [comment]
as implemented, this returns identical results to the script we started with:
```
library(dplyr)
library(SeqArray)
library(blPipeline)

gds <- seqOpen("/nfs/beluga0_home3/DATA/IMPUTED/dbGaP/phs000496/phs000496.dose.filtered.gds")
chr <- c(14,14,19,19)
from_bp <- c(73192712, 73206470, 44908684, 44908822)
to_bp <- c(73192712, 73206470, 44908684, 44908822)
seqSetFilterChrom(gds, chr, from.bp = from_bp, to.bp = to_bp)
dsg_manual <- as.data.frame(seqGetData(gds, "annotation/format/DS"))
rsids <- c("rs63750082", "rs17125721", "rs429358", "rs7412")
colnames(dsg_manual) <- rsids
dsg_manual$sample.id <- seqGetData(gds, "sample.id")
seqResetFilter(gds)

dsg_blp <- extract_dosage_est(gds, chr, from_bp, to_bp)
colnames(dsg_blp) <- c(rsids, "sample.id")

identical(dsg_blp, dsg_manual)
# [1] TRUE
```

**Comment by broomej** [comment]
closed by #6
